Paint example

Bouh opened the pull request:
From https://github.com/GDevelopApp/GDevelop-examples/issues/154

I clean some unused events.
Factorize some of them.
Renamed all resources.
I added the current color in the taskbar.
Added a cross on the current color when we use the eraser.
Fix the draggable behavior name, it was translated into Cyrillic but not revertable in the IDE.

There are still strange pixels on borders when we paint, see this GIF, the border color change:
Any other reviews are welcomed.

![border](https://user-images.githubusercontent.com/1670670/145689719-e68f7e09-bd2e-4bb7-a0ba-ab7c66357372.gif)

Midhil457 commented:
Here is my review:
- Maybe use the slider extension instead of designing from scratch?
- Don't you think the example is sometimes doing some hacky methods to draw thing? I am not saying examples shouldn't be hacky. But, maybe comment out what's going on
- I did see a invalid resource in the resource list. Not sure if it something wrong on my part. But, do fix it if it's reproduceble
- @Bouh, you can add your name to the author field

Midhil457 commented on examples/paint/README.md:
```diff
@@ -0,0 +1 @@
+A regular paint, the ability to draw free-form lines using Shape Painter, the example implements color change using RGB, reading the pixel color using the Read Pixels extension.
```
`A drawing app in which you can draw free form lines.`

And the long description:
` The example uses the shape painter to draw on the screen. And implements a color picker using the Read Pixels extension`

Bouh commented:
> * @Bouh, you can add your name to the author field

No thank you, the original work comes from @Jurfix I let him the honours 💯 

>  I did see a invalid resource in the resource list. Not sure if it something wrong on my part. But, do fix it if it's reproducible

Fixed.

> README.md

Fixed.

Bouh commented:
@arthuro555 Have you an idea of why this happened?


There are still strange pixels on borders when we paint, see this GIF, the border color change:

![border](https://user-images.githubusercontent.com/1670670/145689719-e68f7e09-bd2e-4bb7-a0ba-ab7c66357372.gif)

arthuro555 commented:
That's odd indeed 🤔 I'm not sure why it does that.

Jurfix commented:
> @arthuro555 Have you an idea of why this happened?
> 
> There are still strange pixels on borders when we paint, see this GIF, the border color change:
> 
> ![border](https://user-images.githubusercontent.com/1670670/145689719-e68f7e09-bd2e-4bb7-a0ba-ab7c66357372.gif)

if I'm not mistaken there is an action for drawing a contour along the edge of the window, without this action, the saved line in the sprite will be shifted to zero coordinates
![image](https://user-images.githubusercontent.com/64281856/145731430-d0a8f673-3bef-435f-a527-a25bb9bc0ea6.png)

Bouh commented:
> if I'm not mistaken there is an action for drawing a contour along the edge of the window, without this action, the saved line in the sprite will be shifted to zero coordinates

Ohh nice catch, it was the outline on the PaintBrush object 👍

Bouh commented:
I'm ready to merge, let me know if you see something else to change.

@Midhil457 
>  Don't you think the example is sometimes doing some hacky methods to draw thing? I am not saying examples shouldn't be hacky. But, maybe comment out what's going on

Can you do these comments, please?

If you want I zipped the project from this PR here: [paint.zip](https://github.com/GDevelopApp/GDevelop-examples/files/7700047/paint.zip)

arthuro555 commented:
OOOhhh, that outline 🤦. I thought it was about the outline of the drawn lines changing shapes. Yes, I had told them to add that outline so that the rendered sprite is the correct size, since the size of the rendered sprite is set to the size of the rendered surface, and the outline ensured that this size was constantly the same.

Midhil457 commented:
Did a small cleaup and also commented some events.
Though, I still don't know what this event is for:
![event](https://user-images.githubusercontent.com/73597906/145818122-b7a47a81-0505-4be4-829e-eab7c0b5b1de.png)

I'll approve this after this event is also commented :)

Jurfix commented:
> I'm ready to merge, let me know if you see something else to change.
> 
> @Midhil457
> 
> > Don't you think the example is sometimes doing some hacky methods to draw thing? I am not saying examples shouldn't be hacky. But, maybe comment out what's going on
> 
> Can you do these comments, please?
> 
> If you want I zipped the project from this PR here: [paint.zip](https://github.com/GDevelopApp/GDevelop-examples/files/7700047/paint.zip)

This is a recording of the project from the phone

https://user-images.githubusercontent.com/64281856/145870292-47c4a2b2-218b-4b22-9a82-c7c258b1b1bf.mp4

If you pay attention to this in the original project (which I posted), this problem is not present, it seems to me that you removed the part of the code that is responsible for this, so far I have not understood what exactly, but just keep in mind

Jurfix commented:
But no, I'm sorry, the original project also contains this error, it seemed to me that I managed to fix it, are there any options for how to fix it?

Jurfix commented:
> Did a small cleaup and also commented some events. Though, I still don't know what this event is for: ![event](https://user-images.githubusercontent.com/73597906/145818122-b7a47a81-0505-4be4-829e-eab7c0b5b1de.png)
> 
> I'll approve this after this event is also commented :)

If I'm not mistaken, Arturo wrote why you need to draw a contour around the screen
https://github.com/GDevelopApp/GDevelop-examples/pull/193#issuecomment-991990527

Bouh commented:
I made a fix, see how it's going on my phone, (sorry at the end my phone is lagging.)
https://user-images.githubusercontent.com/1670670/145873898-df4762ab-252b-4f44-926a-9f27c6a11a30.mp4

Jurfix commented:
> I made a fix, see how it's going on my phone, (sorry at the end my phone is lagging.)
> 
>  XRecorder_13122021_201220.mp4

Yes, but still I would like to understand how to fix these line formations with multitouch

https://user-images.githubusercontent.com/64281856/145880807-898be642-1f85-49e7-9736-1e77ecefb2e3.mp4

Jurfix commented:
Or is it okay? In a sense, is this a normal practice for similar applications (painters)?

Bouh commented:
It's a non-expected feature, haha, but still ok to keep it. It's even better, I think.

It happens because the line action is used from the old position (first touch) to the new position (second touch)
![image](https://user-images.githubusercontent.com/1670670/145881596-4ef1d31e-9642-438c-aecd-b13c229554fa.png)

Jurfix commented:
You confused the meaning of green with blue, green should be 166, and blue 35

![Снимок экрана 2021-12-15 230037](https://user-images.githubusercontent.com/64281856/146264582-e22a4c1f-7355-4cf1-b1bd-8b3c6f235026.png)

Bouh commented:
I fixed the colors, If everything is good to you I'll merge for this weekend.

Jurfix commented:
> I fixed the colors, If everything is good to you I'll merge for this weekend.

you can also update the Sprite Snapshot extension to the second version, I saw Arturo released an update with a fix of some bug

arthuro555 commented:
yes if you spam press the screen for a few minutes the game should eventually crash if it uses render to sprite v0.0.1

Bouh commented:
The project will be present in few minutes at:
[https://editor.gdevelop-app.com/?project=https://resources.gdevelop-app.com/examples/paint/paint.json](https://editor.gdevelop-app.com/?project=https://resources.gdevelop-app.com/examples/paint/paint.json)
